### ISSUE TITLE
LLVM Assertion failure with large fixed size array and default implementation

### ISSUE THREAD

**Issue by mrmonday** [rust-lang/rust#21044]
The following code:
```
#[derive(Copy)]
pub struct Foo {
    pub objs: [u8; 512us * 128us],
}
impl ::std::default::Default for Foo {
    fn default() -> Foo {
        unsafe {
            ::std::mem::zeroed()
        }
    }
}
```
When compiled with `rustc --crate-type lib test.rs`, gives the following error:
```
rustc: /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/llvm/include/llvm/CodeGen/SelectionDAGNodes.h:899: llvm::SDValue::SDValue(llvm::SDNode*, unsigned int): Assertion `(!Node || ResNo < Node->getNumValues()) && "Invalid result number for the given node!"' failed.
zsh: abort (core dumped)  rustc /home/robert/rust-dpdk/src/lib.rs --crate-type lib
```
Rust version:
```
rustc 1.0.0-nightly (44a287e6e 2015-01-08 17:03:40 -0800
```
Using Linux x86_64.

**Comment by mrmonday** [comment]
This seems to be fixed in `rustc 1.0.0-nightly (458a6a2f6 2015-01-25 21:20:37 +0000)` --- should I close this issue, or does it need to be marked `E-needstest`?

**Comment by tamird** [comment]
Yep, looks fixed.
```rust
#![crate_type="lib"]

#[derive(Copy)]
pub struct Foo {
    pub objs: [u8; 512 * 128],
}
impl ::std::default::Default for Foo {
    fn default() -> Foo {
        unsafe {
            ::std::mem::zeroed()
        }
    }
}
```
```shell
$ rustc main.rs
main.rs:3:10: 3:14 error: the trait `core::clone::Clone` is not implemented for the type `Foo` [E0277]
main.rs:3 #[derive(Copy)]
                   ^~~~
main.rs:3:10: 3:14 note: in expansion of #[derive_Copy]
main.rs:3:10: 3:14 note: expansion site
error: aborting due to previous error
```
